jobs/build.Jenkinsfile

```diff
@@ -145,6 +145,12 @@ lock(resource: "build-${params.STREAM}") {
 
         try { timeout(time: 240, unit: 'MINUTES') {
 
+        // Add in AWS Build Upload credentials here if they exist. In
+        // the future we might choose to be more granular about when
+        // we load this.
```

Comment:
That's unfortunate. The semantics of `tryWithOrWithoutCredentails` feel funny, but I see how you got here. We fetch things from AWS in a couple of places.

I think eventually we should clean up the `s3_stream_dir` thing though so that it doesn't carry so much weight. We now have a proper config file, so we can make the presence/absence of `s3_bucket` canonical (if it's defined, assume we're pushing; but neuter it if we're in the official Jenkins as a convenience to devs).

Reply:
> I think eventually we should clean up the `s3_stream_dir` thing though so that it doesn't carry so much weight.

agree. I think there are improvements to be made here. It's hard while I'm in the code not to try to do everything, but that would be boiling the ocean.

Still lots of work to do!